tiledb/sm/query/query_condition.cc

```diff
@@ -103,16 +103,6 @@ Status QueryCondition::check(const ArraySchema* const array_schema) const {
           "Clause field name is not an attribute " + field_name);
     }
 
-    if (attribute->var_size()) {
```

Comment:
Modify this check to allow var-sized ASCII types. Error out on all other var-sized types.